test/PSMDST.js

```diff
@@ -0,0 +1,214 @@
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'environment.depth.belowTransducer')
+    delta.updates[0].values[0].value.should.be.closeTo(266.33424, 0.0005)
+  })
+
+  it('0x10 AWA converted', () => {
+    const delta = new Parser().parse(apparentWindAngle)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'environment.wind.angleApparent')
+    delta.updates[0].values[0].value.should.be.closeTo(2.373647783254262, 0.0005)
+  })
+
+  it('0x11 AWS converted', () => {
+    const delta = new Parser().parse(apparentWindSpeed)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'environment.wind.speedApparent')
+    delta.updates[0].values[0].value.should.be.closeTo(0.6173334897244841, 0.0005)
+  })
+
+  it('0x20 STW converted', () => {
+    const delta = new Parser().parse(speedThroughWater)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.speedThroughWater')
+    delta.updates[0].values[0].value.should.be.closeTo(2.6236673313290573, 0.0005)
+  })
+
+  it('0x21 Trip converted', () => {
+    const delta = new Parser().parse(tripMileage)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.trip')
+    delta.updates[0].values[0].value.should.be.closeTo(2674917.68225763, 0.0005)
+  })
+
+  it('0x22 Total log converted', () => {
+    const delta = new Parser().parse(log)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.log')
+    delta.updates[0].values[0].value.should.be.closeTo(4086808.4, 0.5)
+  })
+
+  it('0x25 trip and log converted', () => {
+    const delta = new Parser().parse(tripAndLog)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.trip')
+    delta.updates[0].values[0].value.should.be.closeTo(2665750.28, 0.5)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.log')
+    delta.updates[0].values[1].value.should.be.closeTo(7035562.8, 0.5)
+  })
+
+  it('0x26 STW converted', () => {
+    const delta = new Parser().parse(averageSpeedThroughWater)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.averageSpeedThroughWater')
+    delta.updates[0].values[0].value.should.be.closeTo(22.47, 0.5)
+  })
+
+  it('0x27 Water temperature converted', () => {
+    const delta = new Parser().parse(waterTemperature)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'environment.water.temperature')
+    delta.updates[0].values[0].value.should.be.closeTo(288.9, 0.5)
+  })
+
+  it('0x50 Latitude converted', () => {
+    const delta = new Parser().parse(latitude)
+    delta.updates[0].values.should.contain.an.item.with.property('path', 'navigation.position.latitude')
+    delta.updates[0].values[0].value.should.be.closeTo(33, 0.5)
+  })
+
+  it('0x51 Longitude converted', () => {
+    const delta = new Parser().parse(longitude)
```

Comment:
Oh, this is a problem with the existing Seatalk support. I assume all clients work with the path in spec: `navigation.position` and an object value. This will not be treated as a position, with two separate messages with a non spec path.

Do you know how the data stream works in the real world? Could the parser keep the latest value of one or the other and trigger emit on the other, making sure that the value of the other field is not too old?

Reply:
This is a copy of the ALK.js code. The actual code seems to be ok from https://github.com/SignalK/nmea0183-signalk/pull/156, but the tests may not have followed?

Reply:
tests/ALK.js different https://github.com/SignalK/nmea0183-signalk/blob/master/test/ALK.js#L110-L121

That works, but the position test is not the greatest, because it relies on the tests running in order - they are not independent. All of this is actually just one test https://github.com/SignalK/nmea0183-signalk/blob/master/test/ALK.js#L108-L121 and should be structured as one.

If the tests for ALK and PSMDST are the same, except for the inputs, it would be great to restructure them so that there is only one copy of the tests, running for two sets of test data.

I tried to fix the Gitub test action to run on PRs, could you please rebase on master so that we can see that tests run.